.github/workflows/build-and-test.yml

```diff
@@ -267,12 +267,24 @@ jobs:
     needs: [unittest, integration-tests, lint]
     strategy:
       matrix:
-        binary:
-          - darwin_amd64
-          - darwin_arm64
-          - linux_amd64
-          - linux_arm64
-          - windows_amd64
+        os:
+          - darwin
+          - linux
+          - windows
+        arch:
+          - 386
+          - amd64
+          - arm
+          - arm64
+        exclude:
+          - os: darwin
+            arch: 386
+          - os: darwin
+            arch: arm
```

Comment:
We produce artifacts for this flavor https://github.com/open-telemetry/opentelemetry-collector-releases/blob/49cc73c55020cc3b0a9db9b4aea992037b8037f2/.goreleaser.yaml#L12-L16 so should we avoid ignoring it?

Reply:
Starting with A7 (2013) Apple silicons are 64-bit. In principle it's still possible to run Darwin on other ARM chips but I think it might be rather exotic (perhaps not?) and if it makes sense to build 32-bit Darwin images

Reply:
oh sorry, I misread this as arm64 🤦 makes sense then

Reply:
i'll add a `GOARM` flag 👍